tests/test_snql_api.py

```diff
@@ -415,3 +415,26 @@ def test_escape_edge_cases(self) -> None:
             ),
         )
         assert response.status_code == 200
+
+    def test_multi_table_join(self) -> None:
+        response = self.post(
+            "/events/snql",
+            data=json.dumps(
+                {
+                    "query": f"""
+                    MATCH (e: events) -[grouped]-> (g: groupedmessage),
+                    (e: events) -[assigned]-> (a: groupassignee)
+                    SELECT e.message, e.tags[b], a.user_id, g.last_seen
+                    WHERE e.project_id = {self.project_id}
+                    AND g.project_id = {self.project_id}
+                    AND e.timestamp >= toDateTime('2021-06-04T00:00:00')
+                    AND e.timestamp < toDateTime('2021-07-12T00:00:00')
+                    """,
+                    "turbo": False,
+                    "consistent": False,
+                    "debug": True,
+                }
+            ),
+        )
+
+        assert response.status_code == 200
```

Comment:
SInce you have `"debug": True` here, the response data will have the actual SQL that Clickhouse ran in it. You could so some assertions to make sure that the escaped expression you're expecting is in that SQL. There's also probably tests for the Clickhouse formatter that you should add an extra case too.

Reply:
I would advise against making assertions on the SQL string. We are actually trying to remove them as they are implementation details. Such assertions tend to break for any change that is not really breaking anything but causing the query to be formatted slightly differently. 
See my comment above on testing the specific case by writing unit tests for the formatter instead.
Then for the end to end test, checking that the content of the response is correct will be enough as the details are covered by unit tests.

Regarding tests that need to assert on the AST after query processing (not the case here) I think this approach is better than running assertions on the sql https://github.com/getsentry/snuba/blob/master/tests/datasets/test_sessions_processing.py#L122-L140.

As the test does not depend on formatting.